File: 2022/22/rust/src/main.rs
use rust::Map;

// --- Constants ---
const INPUT_FILE:&'static str = "../input/input.txt";

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // --- Reads and Parse the input file ---
    let text = std::fs::read_to_string(INPUT_FILE)?;
    let i = text.find("\n\n").unwrap();
    let grid = &text[0..i];
    let code = &text[i + 2..];

    // --- Puzzle 1 --
    let mut map = Map::load(grid);
    map.path(code, rust::move_grid);
    let password = map.get_password();
    println!("1. Password {}", password);

    // --- Puzzle 2 ---
    let mut map = Map::load(grid);
    map.path(code, rust::move_cube);
    let password = map.get_password();
    println!("2. Password {}", password);

    Ok(())
}

File: 2022/22/c/main.c
#include <stdio.h>
#include <stdlib.h>
#include <map.h>
#include <lib.h>

// --- Constants ---
#define INPUT_FILE "../input/input.txt"

// --- Main function ---
int main(void) {
	// --- Puzzle 1 ---
	Map* map = malloc(sizeof(Map));
	map_load(map, INPUT_FILE);
	map_path(map, move_grid);
	long total = map_password(map);
	printf("1. Password: %ld\n", total);
	map_free(map); map = NULL;

	// --- Puzzle 2 ---
	map = malloc(sizeof(Map));
	map_load(map, INPUT_FILE);
	map_path(map, move_cube);
	total = map_password(map);
	printf("2. Password: %ld\n", total);
	map_free(map); map = NULL;

	return EXIT_SUCCESS;
}
